acceptance/tests/verify_mountpoints.rb

```diff
@@ -0,0 +1,41 @@
+# This test is intended to demonstrate that mount resource can mount tmpfs directory
+# and the mount facter mountpoints should show the mount on tmpfs
+
+test_name "FACT-1502 - C98163 mountpoints fact should show mounts on tmpfs" do
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  confine :except, :platform => 'windows'
+
+  agents.each do |agent|
+    dir = '/tmp/tempdir'
+    on(agent, "mkdir -p #{dir}")
+    manifest  = 'mount_manifest.pp'
+
+    teardown do
+      on(agent, "umount #{dir}")
+      on(agent, "rm -rf #{dir} #{manifest}")
+    end
+
+    create_remote_file(agent, manifest, <<-FILE)
+      mount {"#{dir}":
+        ensure  => mounted,
+        options => 'noexec',
+        fstype  => 'tmpfs',
+        device  => 'tmpfs',
+        atboot  => true,
+      }
+    FILE
+
+    step "Apply the manifest to mount" do
+      on(agent, puppet("apply #{manifest}")) do
+        assert_no_match(/Error/, stdout, "Unexpected error was detected!")
```

Comment:
I think you should be able to add `:acceptable_exit_codes => [0]` as a third argument to the on(agent, ...) call. It will fail if it gives any non-zero exit code.